gpkit/small_classes.py

```diff
@@ -64,6 +64,16 @@ def __call__(self):
         return self.start
 
 
+class SolverLog(list):
+    "Adds a `write` method to list so it's file-like and can replace stdout."
+
+    def write(self, writ):
+        if writ != "\n":
+            if writ and writ[-1] == "\n":
+                writ = writ[:-1]
+            self.append(writ)
+
```

Comment:
Did you mean to say something else?

Mine won't append `""`.

Yours actually will append `""`...